### PR TITLE
Adjust GrandTotal Check in isExpressCheckoutActive Method

### PR DESCRIPTION
**Pull request by ahmardi**
Refactors the isExpressCheckoutActive method to hide the checkout button if the GrandTotal is zero.

### CODE REVIEW

**Review by Skullsneeze — APPROVED** [review]

File: ViewModel/ExpressCheckoutButton.php [hunk 1]
```diff
@@ -84,7 +84,11 @@ public function shouldRender(): bool
      */
     public function isExpressCheckoutActive(): bool
```
Comment (sameh-amwal) [inline ViewModel/ExpressCheckoutButton.php:85]: why is the method called `isExpressCheckoutActive` when it is really semantically `isExpressCheckoutInactive`??

Comment (ahmardi) [reply ViewModel/ExpressCheckoutButton.php:85]: this old method, we can change the name if not reflected.
@Skullsneeze ?

**Review by sameh-amwal — APPROVED** [review]

File: ViewModel/ExpressCheckoutButton.php [hunk 2]
```diff
@@ -76,15 +76,19 @@ public function __construct(
      */
     public function shouldRender(): bool
     {
-        return $this->isExpressCheckoutActive();
+        return $this->isExpressCheckoutDisabled();
```
Comment (sameh-amwal) [inline ViewModel/ExpressCheckoutButton.php:79]: this still doesn't make sense. `ShouldRender` should be when expressCheckout is enabled not when it is disabled. right?!

File: ViewModel/ExpressCheckoutButton.php [hunk 3]
```diff
@@ -76,15 +76,19 @@ public function __construct(
...
     }
 
     /**
      * @return bool
      */
-    public function isExpressCheckoutActive(): bool
+    public function isExpressCheckoutDisabled(): bool
```
Comment (Skullsneeze) [inline ViewModel/ExpressCheckoutButton.php:85]: Using the `!()` statment means you do a negative check on whatever is between the parenthesis.

In this case we check:

- Is the module disabled (`!$this->config->isActive()`)
- Is the express checkout disabled (`!$this->config->isExpressCheckoutActive()`)
- Is the current current code not the allowed currency code (`$this->storeManager->getStore()->getCurrentCurrencyCode() != self::AMWAL_CURRENCY`)
- Is the Quote total exactly 0 (`$this->checkoutSessionFactory->create()->getQuote()->getGrandTotal() == 0`)

If any of these statements are true, it means the express checkout is disabled, so when our check simplified is:
`!$isCheckoutDisabled` 

Meaning the naming of the function would more logically be `isExpressCheckoutActive` or `isExpressCheckoutEnabled`

**Review by Skullsneeze — CHANGES_REQUESTED** [review]

**Review by sameh-amwal — APPROVED** [review]

**Pull request merged**